script/newsbar/rolling.js

```diff
@@ -0,0 +1,99 @@
+import { getQuerySelector, getQuerySelectorAll } from "../../utils/js/getElements.js";
+import { fetchData } from "../../utils/js/getJson.js";
+
+let intervalFirstNewsbar;
+let intervalSecondNewsbar;
+
+// 최신 헤드라인 뉴스 자동 롤링 시작
+function startRolling() {
+  intervalFirstNewsbar = setInterval(() => {
+    rollingInterval('first');
+  }, 5000);
+
+  intervalSecondNewsbar = setInterval(() => {
+    setTimeout(() => {
+      rollingInterval('second');
+    }, 1000)
+  }, 5000);
+}
+
+// 최신 헤드라인 뉴스 자동 롤링 중지
+function stopRolling() {
+  clearInterval(intervalFirstNewsbar);
+  clearInterval(intervalSecondNewsbar);
+}
+
+// 헤드라인 뉴스 자동 롤링 영역 마우스 이벤트
+function mouseEventRolling(state) {
+  const headlineNews = getQuerySelectorAll(document, `.newsbar-content-container-${state} li`);
+
+  headlineNews.forEach((elem) => {
+    elem.addEventListener('mouseover', () => {
+      elem.style.textDecoration = "underline";
+      elem.style.cursor = "pointer";
+      stopRolling();
+    })
+
+    elem.addEventListener('mouseout', () => {
+      elem.style.textDecoration = "none";
+      elem.style.cursor = "auto";
+      startRolling();
+    })
+  })
+}
+// 헤드라인 뉴스 데이터 html에 입력
+function putNewsHeadline(state, headlineTitleArr) {
+  let headlineSrc = "";
+  headlineTitleArr.forEach((elem, id) => {
+    switch (id) {
+      case 0:
+        headlineSrc += `<li class="current"><a>${elem}</a></li>`;
+        break;
+      case 1:
+        headlineSrc += `<li class="next"><a>${elem}</a></li>`;
+        break;
+      case 4:
+        headlineSrc += `<li class="prev"><a>${elem}</a></li>`;
+        break;
+      default:
+        headlineSrc += `<li class="next"><a>${elem}</a></li>`;
+        break;
+    }
+  });
+  const headlineContainer = getQuerySelector(document, `.newsbar-content-container-${state}`);
+  headlineContainer.innerHTML = headlineSrc;
+  mouseEventRolling(state);
+}
+
+// 헤드라인 뉴스 데이터 받아오기
+async function getNewsHeadline() {
+  const headlinePath = await fetchData("../assets/data/newsTitle.json");
+  const headlineTitleFirst = headlinePath.titleFirst.map((elem) => {
+    return elem.name;
+  })
```

Comment:
map 함수에서 중괄호를 사용하지 않으면 해당 값이 자동으로 `return` 이 됩니다. 따라서 만약 내부로직이 한줄 코드 일 경우 아래와 같이 줄일수도 있을거 같아요!

```suggestion
  const headlineTitleFirst = headlinePath.titleFirst.map((elem) => elem.name)
```

Reply:
중괄호를 제외하고 이렇게 쓸수도 있군요 !! 몰랐던 문법 알려주셔서 감사합니다 ~~